### DISCUSSION TITLE
LlamaCppModel training

### DISCUSSION THREAD

**Discussion by orcadevelopmenteers** [oobabooga/text-generation-webui#1950] [Q&A]
Can we expect support for LlamaCppModel training? That would be really great.

**Comment by ahsueh1996** [comment]
Hey @orcadevelopmenteers  is this the similar error that you are encountering? Is this a Windows specific issue? or is there a difference between "llama" and "llamaCppModel"?
![image](https://github.com/oobabooga/text-generation-webui/assets/9125820/70f53fa1-8551-4c55-a4fa-1dfaf6e7845f)


Update: I did a bit of homework and followed the discussion in https://github.com/oobabooga/text-generation-webui/discussions/1199 . I was successful training the llama-7b-hf model. Still very curious what's the difference between llama and llamaCppModel. I'd be happy to help write some code to contribute to the project... just need some hints if anyone knows anything about their differences! TY~

**Reply by mindobjects** [reply]
@ahsueh1996 I have the same problem here.
I looking for a model with llama for CPU (no GPU). 
Do you have any advise ?

+1 for the LlamaCppModel support

Thanks.

**Reply by ahsueh1996** [reply]
Hey @mindobjects it was a while since I looked at this but I think there was a difference between the models that I wanted to try. 

I believe this https://huggingface.co/decapoda-research/llama-7b-hf/tree/main was the first model I got working with the suggestions in https://github.com/oobabooga/text-generation-webui/discussions/1199. 

Then I tried https://huggingface.co/TheBloke/wizard-vicuna-13B-GGML/blob/previous_llama/wizard-vicuna-13B.ggml.q4_2.bin but the issue became my GPU memory so I stopped there for the next while. To get this one working involved some tinkering around with the right tokenizer or loading... I forget a lot of the details though! 

There are some notes I took and it'll take me a little time to review what I did. If you're interested in discussing we can jump on a call to see if there's anything I can help you out with.
